src/core/connection.c

```diff
@@ -2465,8 +2494,178 @@ QuicConnValidateTransportParameterCIDs(
     return TRUE;
 }
 
+QUIC_STATUS
+QuicConnProcessPeerVersionNegotiationTP(
+    _In_ QUIC_CONNECTION* Connection
+    )
+{
+    QUIC_STATUS Status;
+    if (QuicConnIsServer(Connection)) {
+        //
+        // Check whether version is in (App-specified) list of supported versions
+        //
+        uint32_t SupportedVersionsLength = 0;
+        const uint32_t* SupportedVersions = NULL;
+        if (MsQuicLib.Settings.IsSet.DesiredVersionsList) {
+            SupportedVersionsLength = MsQuicLib.Settings.DesiredVersionsListLength;
+            SupportedVersions = MsQuicLib.Settings.DesiredVersionsList;
+        } else {
+            SupportedVersionsLength = ARRAYSIZE(DefaultSupportedVersionsList);
+            SupportedVersions = DefaultSupportedVersionsList;
+        }
+
+        uint32_t CurrentVersionIndex = 0;
+        for (; CurrentVersionIndex < SupportedVersionsLength; ++CurrentVersionIndex) {
+            if (Connection->Stats.QuicVersion == SupportedVersions[CurrentVersionIndex]) {
+                break;
+            }
+        }
+        if (CurrentVersionIndex == SupportedVersionsLength) {
+            //
+            // Current version not supported, start incompatible version negotiation.
+            //
+            CXPLAT_DBG_ASSERTMSG(FALSE,"Incompatible Version Negotation should happen in binding layer");
```

Comment:
Space after comma. Also, the comment here seems confusing mixed with the assert. Nothing should be "started" after a possible assert IMO.

Reply:
I'll move the comment closer to the return value. The assert is to help developers not make any mistakes, and the if-check is to make sure production systems aren't brought down by hitting this path.